packages/core/src/libraries/hook/index.ts

```diff
@@ -88,51 +117,45 @@ export const createHookLibrary = (queries: Queries) => {
       userIp,
       user: user && pick(user, ...userInfoSelectFields),
       application: application && pick(application, 'id', 'type', 'name', 'description'),
-    } satisfies Omit<HookEventPayload, 'hookId'>;
+    } satisfies InteractionHookEventPayloadWithoutHookId;
 
-    await Promise.all(
-      rows.map(async ({ id, config, signingKey }) => {
-        consoleLog.info(`\tTriggering hook ${id} due to ${hookEvent} event`);
-        const json: HookEventPayload = { hookId: id, ...payload };
-        const logEntry = new LogEntry(`TriggerHook.${hookEvent}`);
+    await sendWebhooks(rows, payload);
+  };
 
-        logEntry.append({ hookId: id, hookRequest: { body: json } });
+  /**
+   * Trigger data hooks with the given data mutation context. All context objects will be used to trigger hooks.
+   */
+  const triggerDataHooks = async (hooks: DataHookContextManager) => {
+    if (hooks.contextArray.length === 0) {
+      return;
+    }
 
-        // Trigger web hook and log response
-        await sendWebhookRequest({
-          hookConfig: config,
-          payload: json,
-          signingKey,
-        })
-          .then(async (response) => {
-            logEntry.append({
-              response: await parseResponse(response),
-            });
-          })
-          .catch(async (error) => {
-            logEntry.append({
-              result: LogResult.Error,
-              response: conditional(
-                error instanceof HTTPError && (await parseResponse(error.response))
-              ),
-              error: conditional(error instanceof Error && String(error)),
-            });
-          });
+    const found = await findAllHooks();
 
-        consoleLog.info(
-          `\tHook ${id} ${logEntry.payload.result === LogResult.Success ? 'succeeded' : 'failed'}`
+    await Promise.all(
```

Comment:
concern for the concurrency issue as above. i'm thinking to flat map all promises instead of limit concurrency separately

Reply:
updated